genesis/process/shardGenesisBlockCreator.go

```diff
@@ -494,3 +511,35 @@ func executeDelegation(
 
 	return delegationProcessor.ExecuteDelegation()
 }
+
+func incrementCrossShardDelegators(processors *genesisProcessors, arg ArgsGenesisBlockCreator) (int, error) {
```

Comment:
1. The name of the function is misleading.
I would expect a crossShardDelegator++
what it does besides returing the number of delegations, is incrementing the nonces for each of the accounts that do delegation, for every delegation they do cross shad. 

so I would rename the function to something like updateNoncesForCrossShardDelegations

2. What happens with the nonces for accounts that do delegations in the same shard?

Reply:
renamed to incrementNoncesForCrossShardDelegations (as it does an increment). The reason this is not called for the intra shard delegators is that the tx processor does that automatically when executing the transaction